dotnet/src/Connectors/Connectors.Onnx/OnnxRuntimeGenAIPromptExecutionSettings.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Text.Json.Serialization;
+
+namespace Microsoft.SemanticKernel.Connectors.Onnx;
+
+/// <summary>
+/// OnnxRuntimeGenAI Execution Settings.
+/// </summary>
+public sealed class OnnxRuntimeGenAIPromptExecutionSettings : PromptExecutionSettings
+{
+    /// <summary>
+    /// Convert PromptExecutionSettings to OnnxRuntimeGenAIPromptExecutionSettings
+    /// </summary>
+    /// <param name="executionSettings"></param>
+    /// <returns></returns>
+    public static OnnxRuntimeGenAIPromptExecutionSettings FromExecutionSettings(PromptExecutionSettings? executionSettings)
+    {
+        switch (executionSettings)
+        {
+            case OnnxRuntimeGenAIPromptExecutionSettings settings:
+                return settings;
+            default:
+                return new OnnxRuntimeGenAIPromptExecutionSettings();
+        }
+    }
+
+    [JsonPropertyName("top_k")]
+    public int TopK { get; set; } = 50;
```

Comment:
Does the underlying genapi have defaults for all these? Wondering if we should make these all nullable, and then only transfer the setting if explicitly set?